predicators/spot_utils/utils.py

```diff
@@ -162,11 +163,16 @@ def sample_move_offset_from_target(
         rot = angle + np.pi if angle < 0 else angle - np.pi
         cand_geom = Rectangle.from_center(x, y, robot_geom.width,
                                           robot_geom.height, rot)
-        # Check for out-of-bounds.
-        oob = False
-        for cx, cy in cand_geom.vertices:
-            if cx < min_x or cy < min_y or cx > max_x or cy > max_y:
-                oob = True
+        # Check for out-of-bounds. To do this, we're looking for
+        # one allowed region where all four points defining the
+        # robot will be within the region in the new pose.
+        oob = True
+        for region in allowed_regions:
+            for cx, cy in cand_geom.vertices:
+                if region.find_simplex(np.array([cx, cy])) < 0:
```

Comment:
I'm not familiar with this algorithm. Did you see this used for this purpose somewhere else? It sounds reasonable but maybe slow and approximate?

Reply:
Got it from [here](https://stackoverflow.com/questions/16750618/whats-an-efficient-way-to-find-if-a-point-lies-in-the-convex-hull-of-a-point-cl).

I was worried about it being slow/approximate, but integration testing seems to show it's relatively fast and seems accurate enough for our purposes. I believe it gets much worse when the points are high dimensional and there are a lot of them, but for a few points in 2D, it should be pretty fine.